launch_ros/launch_ros/actions/node.py

```diff
@@ -244,6 +244,13 @@ def parse(cls, entity: Entity, parser: Parser):
         node_name = entity.get_attr('name', optional=True)
         if node_name is not None:
             kwargs['node_name'] = parser.parse_substitution(node_name)
+            # Undo the parent action parsing 'name' as the process name
+            # For node actions, use 'exec-name' to rename the process
+            if 'name' in kwargs:
+                del kwargs['name']
+        exec_name = entity.get_attr('exec-name', optional=True)
+        if exec_name is not None:
+            kwargs['name'] = parser.parse_substitution(exec_name)
```

Comment:
@jacobperron hmm, while this is a fine patch, it grows the gap between Python and markup-based launch files. This is something we've discussed with @ivanpauno several times, how detrimental to standardization is the fact that launch frontend parsing support allows one to depart from the original action "signature". IMHO, we should apply this exact same change to the `Node` action constructor itself and then parsing function should just use it. WDYT?

Reply:
The problem with changing the `Node` signature is that it breaks API in a confusing way:
if somebody was using `name` to specify the executable name, they will have a surprise when that's now used as node name.
I think will should keep `XML` attributes as close as possible to `Python` constructor arguments.

Reply:
That is also true. But doing so in XML/YAML only but not in Python makes it worst IMO.

Reply:
I agree that this change should happen in the Node constructor, keeping the signature the same as the front-end. I'll do this if there's a consensus to go ahead with this change.

> The problem with changing the Node signature is that it breaks API in a confusing way:
if somebody was using name to specify the executable name, they will have a surprise when that's now used as node name.

Yeah, this is a problem. IMO, it seems like a good thing to change long term. In the short-term, we can document the behavior change in the release notes and on whatever documentation exists for launch_ros (note, besides the API docstrings, I only found mention of the "name" attribute in the [migration guide](https://index.ros.org/doc/ros2/Tutorials/Launch-files-migration-guide/#node) and it is already assuming the behavior I expect).

I also understand if there's consensus to leave the implementation as-is in order to avoid breaking the existing behavior.

Reply:
I think that the API break is reasonable.
To avoid breaking user launch files, `node_name` can be deprecated with a warning.
The users that were using `name` will have to check the release notes to understand the problem (though I expect not much users using it).
We should also merge the argument name change together with a PR updating the release notes. If not, we will forget later.

If @wjwwood and @hidmic are also ok with breaking API, I think it's fine.

Reply:
👍 to breaking API to make it make sense, and 👍 for updating the documentation and release notes simultaneously.

Reply:
Thanks for the input, all. I'm on it :sunglasses: